framework/doc/content/application_usage/mesh_block_type.md

```diff
@@ -0,0 +1,173 @@
+# Mesh
+
+## Creating a Mesh
+
+- For complicated geometries, we generally use CUBIT from Sandia National Laboratories.
+- CUBIT can be licensed from CSimSoft for a fee depending on the type of organization you work for.
+- Other mesh generators can work as long as they output a file format that libMesh reads (next section).
+- If you have a specific mesh format you like, we can take a look at adding support for it to libMesh.
+
+### FileMesh
+
+- `FileMesh` is the default type.
+- MOOSE supports reading and writing a large number of formats and could be extended to read more.
+
+| Extension   | Description                              |
+| :-          | :-                                       |
+| .dat        | Tecplot ASCII file                       |
+| .e, .exd    | Sandia's ExodusII format                 |
+| .fro        | ACDL's surface triangulation file        |
+| .gmv        | LANL's GMV (General Mesh Viewer) format  |
+| .mat        | Matlab triangular ASCII file (read only) |
+| .msh        | GMSH ASCII file                          |
+| .n, .nem    | Sandia's Nemesis format                  |
+| .plt        | Tecplot binary file (write only)         |
+| .node, .ele; .poly | TetGen ASCII file (read; write)   |
+| .inp        | Abaqus .inp format (read only)           |
+| .ucd        | AVS's ASCII UCD format                   |
+| .unv        | I-deas Universal format                  |
+| .xda, .xdr  | libMesh formats                          |
+| .vtk, .pvtu | Visualization Toolkit                    |
```

Comment:
Could each of these format type explanations under the description heading be updated to include hyperlinks to the associated websites?

Reply:
Good suggestion @sapitts, @leorachapuis is going to add some links for the different formats where available.

Reply:
Awesome! Thanks @leorachapuis!